src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/PkcsFormatReader.cs

```diff
@@ -218,29 +218,29 @@ internal static bool TryReadPkcs7Pem(SafeBioHandle bio, out List<ICertificatePal
             return true;
         }
 
-        internal static bool TryReadPkcs12(byte[] rawData, string password, out ICertificatePal certPal)
+        internal static bool TryReadPkcs12(byte[] rawData, object password, out ICertificatePal certPal)
```

Comment:
I really, really don't like this.  Especially because I think we marshal this string out quite a lot during the "is it this format?" cascade, which can be really expensive for SecureString.

How about we turn it into the pointer very early in the process and pass the IntPtr through?
Better yet, make it a SafeHandle type.